src/storage/blocksstable/ob_lob_merge_writer.cpp

```diff
@@ -410,10 +413,15 @@ int ObLobMergeWriter::write_lob_obj(
 int ObLobMergeWriter::copy_row_(const ObStoreRow& row)
 {
   int ret = OB_SUCCESS;
-  result_row_ = row;
-  result_row_.row_val_.cells_ = reinterpret_cast<ObObj*>(buffer_);
-  result_row_.row_val_.count_ = row.row_val_.count_;
-  MEMCPY(buffer_, row.row_val_.cells_, sizeof(ObObj) * row.row_val_.count_);
+  int64_t request_count = data_store_desc_->row_column_count_;
+  if OB_FAIL(buffer_.reserve(request_count)) {
+    STORAGE_LOG(WARN, "fail to reserve memory for buffer_, ", K(ret));
+  } else {
+    result_row_ = row;
+    result_row_.row_val_.cells_ = buffer_.get_buf();
+    result_row_.row_val_.count_ = row.row_val_.count_;
+    MEMCPY(buffer_.get_buf(), row.row_val_.cells_, sizeof(ObObj) * row.row_val_.count_);
```

Comment:
add check ensure buffer is enough

Reply:
I change the reserved buffer size to row.row_val_.count_ which is always an enough buffer.